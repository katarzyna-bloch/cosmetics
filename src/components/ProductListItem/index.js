import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'
import media from '../../utils/media'
import { Link as RouterLink } from 'react-router-dom'
import Price from '../Price'

const Link = styled(RouterLink)`
  text-decoration: none;
  color: black;
  flex: 0 25%;
  border: 1px solid transparent;
  box-sizing: border-box;
  border-radius: 5px;

  &:hover {
    border-color: lightgray;
  }

  ${media.desktop} {
    flex: 0 33%;
  }

  ${media.tablet} {
    flex: 0 50%;
  }

  ${media.phone} {
    flex: 1 100%;
  }
`

const Wrapper = styled.div`
  text-align: center;
`

const Image = styled.img`
  height: 180px;
  padding: 15px 20px 0px;

  ${media.tablet} {
    height: 140px;
  }
`

const Name = styled.div`
  text-align: center;
  padding: 5px;
`

const ProductListItem = ({ id, name, img, oldPrice, price, hasDiscount }) => (
  <Link to={`products/${id}`}>
    <Wrapper>
      <Image src={img} alt="cosmetic" />
      <Name>{name}</Name>
      <Price hasDiscount={hasDiscount} oldPrice={oldPrice} price={price} />
    </Wrapper>
  </Link>
)

ProductListItem.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  oldPrice: PropTypes.string,
  hasDiscount: PropTypes.bool,
}

export default ProductListItem
