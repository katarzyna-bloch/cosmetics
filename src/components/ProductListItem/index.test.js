import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import ProductListItem from "."

it("should render correctly", () => {
  const props = {
    name: "Test product!",
    img: "img.jpg",
    price: "15z≈Ç",
    id: "1",
  }
  const wrapped = shallow(<ProductListItem {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
