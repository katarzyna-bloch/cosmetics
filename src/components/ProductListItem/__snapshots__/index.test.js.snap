// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Styled(Link)
  to="products/1"
>
  <styled.div>
    <styled.img
      alt="cosmetic"
      src="img.jpg"
    />
    <styled.div>
      Test product!
    </styled.div>
    <styled.div>
      15z≈Ç
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
